src/cpp/fastdds/publisher/PublisherImpl.cpp

```diff
@@ -247,23 +245,18 @@ DataWriter* PublisherImpl::create_datawriter(
     // Check netmask filtering preconditions
     if (nullptr != rtps_participant())
     {
-        fastrtps::rtps::RTPSParticipantImpl* rtps_impl = fastrtps::rtps::RTPSDomainImpl::find_local_participant(
-            rtps_participant()->getGuid());
-        if (nullptr != rtps_impl)
+        std::vector<fastdds::rtps::TransportNetmaskFilterInfo> netmask_filter_info =
+                rtps_participant()->netmask_filter_info();
```

Comment:
Same for the subscriber
```suggestion
                rtps_participant_->netmask_filter_info();
```